### ISSUE TITLE
[11.0] calendar.event (Meeting) sorting does not work

### ISSUE THREAD

**Issue by katyukha** [odoo/odoo#23599]
### Impacted versions:
11.0

### Steps to reproduce:
1. go to *Calendar* app
2. Open any *event* and set field *location* to any value (for example 'test')
3. Go to *list* view
4. Sort by location

### Current behavior:
```
Traceback (most recent call last):
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/http.py", line 647, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/http.py", line 689, in dispatch
    result = self._call_function(**self.params)
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/http.py", line 933, in __call__
    return self.method(*args, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/addons/web/controllers/main.py", line 872, in search_read
    return self.do_search_read(model, fields, offset, limit, domain, sort)
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/addons/web/controllers/main.py", line 894, in do_search_read
    offset=offset or 0, limit=limit or False, order=sort or False)
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/models.py", line 4139, in search_read
    records = self.search(domain or [], offset=offset, limit=limit, order=order)
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/addons/calendar/models/calendar.py", line 1676, in search
    events = self.browse(events.get_recurrent_ids(args, order=order))
  File "/home/odoo/runbot/extra/runbot/static/build/310161-11-0-f02cd2/odoo/addons/calendar/models/calendar.py", line 1154, in get_recurrent_ids
    return [r['id'] for r in sorted(result_data, key=key)]
TypeError: unorderable types: str() < bool()
```

### Expected behavior:
Sorted list of events

### Video/Screenshot link (optional):
![Events list before click on sort by location](https://user-images.githubusercontent.com/1129820/37283355-5dba201c-2600-11e8-9feb-4cfa8e182ca2.png)
![Error message after click on sort by location](https://user-images.githubusercontent.com/1129820/37283388-78b27eaa-2600-11e8-8894-2c402d3ac7d9.png)

### Same when sorting on many2one fields (see Responsible field)
![Events list with one record without responsible user](https://user-images.githubusercontent.com/1129820/37283789-f3a6f572-2601-11e8-9385-f355ace41bb9.png)
![Error on attempt to sort by field 'responsible user'](https://user-images.githubusercontent.com/1129820/37283821-0d588e72-2602-11e8-8e36-6e38c5462cbf.png)

**Comment by Elkasitu** [comment]
@xmo-odoo Looks like something to do with tools.Reverse and some generators of yours :laughing:

**Comment by burhan-vakharia** [comment]
I can confirm this issue. I could reproduce with the exact same steps.

**Comment by sswapnesh** [comment]
@Elkasitu  This has been fixed in b6c962e62a7a47240d0091f88fc2ad4359179845

**Comment by Yenthe666** [comment]
Indeed - thanks for the headsup @sswapnesh
